src/adapters/geocoder.js

```diff
@@ -0,0 +1,79 @@
+import ajax from 'src/libs/ajax';
+import nconf from '@qwant/nconf-getter';
+import QueryContext from 'src/adapters/query_context';
+import BragiPoi from 'src/adapters/poi/bragi_poi';
+import Intention from './intention';
+
+const serviceConfigs = nconf.get().services;
+const geocoderConfig = serviceConfigs.geocoder;
+const geocoderFocusPrecision = geocoderConfig.focusPrecision;
+
+const bragiCache = {};
+
+function roundWithPrecision(value, precision) {
+  const rounded = Math.round(value * (1 / precision)) * precision;
+  return rounded.toFixed(3);
```

Comment:
ok, we can even use `digits` parameter with default value of 3

Reply:
also, maybe we should move this reusable function in a lib

Reply:
Why not if we extract that to a shared function, here I just copy-pasted the most part of the bragi_poi file.